### DISCUSSION TITLE
Return to DOSBox-X prompt after exiting Windows 9x?

### DISCUSSION THREAD

**Discussion by emendelson** [joncampbell123/dosbox-x#4421] [Q&A]
When running Windows 9x under DOSBox-X, is it possible to return to the DOSBox-X prompt after exiting Windows? I want to perform a few operations automatically after exiting Windows 9x, but I can't see any way to make DOSBox-X stay open after exiting the Windows guest. Am I missing something obvious?

**Comment by rderooy** [comment]
No, because you booted a guest OS, you cannot return to the DOSBox-X prompt on exit.

**Comment by emendelson** [comment]
Do you think it might be possible to add an option to the BOOT command, something like `/exittoprompt`? If so, I'll file a feature request? It would be good to be able to run some commands after the guest OS shuts down.

**Comment by rderooy** [comment]
Well, it would not be as much as an exit, but rather a reboot-to. But that could cause issues also as a guest OS such as Windows 9x often requires reboots for small changes, and this would interrupt that.

**Comment by emendelson** [comment]
Understood. It would be useful to have an [autoexec] that can have additional commands after `boot C:`.

**Reply by Torinde** [reply]
You could make some logic like below?
- [Win9x guest] instruct users to shutdown via BAT file that creates somewhere a file "EXECUTE-ON-NEXT-REBOOT" - or reversely put that in its autoexec.bat and instruct users to delete the file if they are rebooting/restarting due to some software installation or another reason where your commands should not be executed.
- [DOSbox] restart/reset
- [autoexec] if "EXECUTE-ON-NEXT-REBOOT" exists, then execute your commands and delete "EXECUTE-ON-NEXT-REBOOT"

**Comment by rderooy** [comment]
If this were to be implemented, i'm thinking it should perhaps be a "reboot-to-shell-on-halt". Such that when a guest OS issues a halt, dosbox-x will shutdown the guest OS as instructed, but do a reboot to the shell instead of a shutdown. Obviously the next question would be what batch file should be run, as it cannot be the normal [autoexec] section.

This would prevent issues with win95 wanting to reboot for every little change.

**Comment by emendelson** [comment]
Perhaps an [autoexec-restart] section might be possible to implement? Or something like this in [autoexec]:

```
[autotexec]
if restart go :restart
....
....
go :end-it
:restart
....
....
:end-it
exit
```

**Comment by cowarlydragon** [comment]
As an extension/alternative to this, it might be nice to have a set of dos / dosbox command to "cleanup" post windows exit. 

Like... push changes to FAT drive back to host folder, but at this point the guest OS has closed and that emulation may be kapoof.
